### PR TITLE
Some improvements on malloc calls

### PR DESCRIPTION
**Pull request by zadlg**
Almost all `malloc` calls are checked.
Plus:

rewrite:
```
void* p = malloc(..);
if (!p) {
```
by:
```
if (!(p = malloc(..))) {
```

*Edit*: my clang-format removed some trailing spaces

### CODE REVIEW

File: libr/asm/p/asm_sh.c [hunk 1]
```diff
@@ -44,9 +44,9 @@ static int buf_fprintf(void *stream, const char *format, ...) {
 	int ret;
 	if (!buf_global)
 		return 0;
-	tmp = malloc (strlen (format)+strlen (buf_global)+2);
-	if (!tmp)
+	if (!(tmp = malloc (strlen (format)+strlen (buf_global)+2))) {
 		return 0;
+	}
```
Comment (radare) [inline libr/asm/p/asm_sh.c]: just use r_str_newf

**Comment by zadlg** [comment]
what is that ?
I've just rewritten some parts, without really knowing the whole function, I've just checked for freeing variables and being consistent with the return value if any

**Comment by radare** [comment]
Please rebase

**Comment by zadlg** [comment]
done

**Comment by XVilka** [comment]
There are a lot of not yours commits, can you please do a clean rebase?

**Comment by zadlg** [comment]
sorry about that.

I've cleaned up the mess

**Comment by xarkes** [comment]
It's unrelated, there's something wrong on master.

**Comment by xarkes** [comment]
It's been fixed a few minutes ago, rebase again and it should become green.

**Comment by xarkes** [comment]
```
[XX] db/cmd/cmd_long_string pD@x:90  3
$ r2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Q -e asm.cmt.right=false -i /var/folders/bb/n7t3rs157850byt_jfdcq9k80000gn/T/tmp-28038YS83gsVpuDsQ.tmp ../bins/mach0/bin_longstring
s 0x10000088a; pd 3~?string
-4
+3
```

**Comment by zadlg** [comment]
hmm weird, I'm gonna check my changes on mach0

**Comment by zadlg** [comment]
Okay, I couldn't see something wrong in my changes, someone has an idea why only one test fails on only mac+clang conf?

**Comment by radare** [comment]
try rebasing, now master should be green and stable

may be undefined behaviour, but the changes are so large to review them all at once

**Comment by zadlg** [comment]
wtf, now the 4 tests failed, before it was the last only
I'm gonna debug that.

I know the changes are large, but it just basic rewrites and some added checks for malloc results

**Comment by Maijin** [comment]
I see only one test failing

**Comment by zadlg** [comment]
I meant the 4 Travis jobs, sorry

**Comment by xarkes** [comment]
It's boring because master is red. But I think it's good to merge.

**Comment by zadlg** [comment]
should be okay, it's green

**Comment by radare** [comment]
I dont see why inlining assignments in the conditional statement improves the reading in any way. Also thats not specified in the coding style document.

I also miss many spaces before ( and there are many places where this malloc can be improved with rstrnewf, rnew0 or calloc

**Comment by zadlg** [comment]
alright, fair enough

**Pull request closed without merging**